File: FactFront-main/src/types/entra-user.ts
export interface EntraUser {
  id: string;
  displayName: string | null;
  userPrincipalName: string | null;
  mail: string | null;
  jobTitle: string | null;
  accountEnabled: boolean;
  /** "Member" for internal employees, "Guest" for invited partners. */
  userType: string | null;
  roles: string[];
}

export interface InviteRequest {
  email: string;
  displayName?: string;
  roles?: string[];
}

File: FactBack-main/src/main/java/fr/alb/platform/admin/dto/EntraUser.java
package fr.alb.platform.admin.dto;

import java.util.List;

/**
 * Flat view of an Entra user as needed by the admin UI. Derived from the
 * Graph {@code /users} payload plus a pre-computed list of roles the user
 * holds (mapped from Entra group memberships via
 * {@code azure.ad.group.mapping.*}).
 */
public record EntraUser(
        String id,
        String displayName,
        String userPrincipalName,
        String mail,
        String jobTitle,
        boolean accountEnabled,
        String userType,
        List<String> roles
) {}
